CONTRIBUTING.rst

```diff
@@ -0,0 +1,195 @@
+
+.. code-block:: bash
+
+    git clone <insert/your/fork/url/here>
+    cd rascaline
+
+    # setup the local repository so that the master branch tracks changes in
+    # the main repository
+    git remote add upstream https://github.com/Luthaf/rascaline/
+    git fetch upstream
+    git branch master --set-upstream-to=upstream/master
+
+Once you get the code locally, you will want to run the tests to check
+everything is working as intended. See the next section on this subject.
+
+If everything is working, you can create your own branches to work on your
+changes:
+
+.. code-block:: bash
+
+    git checkout -b <my-branch-name>
+    # code code code
+
+    # push your branch to your fork
+    git push -u origin <my-branch-name>
+    # follow the link in the message to open a pull request (PR)
+
+.. _create a fork: https://docs.github.com/en/github/getting-started-with-github/fork-a-repo
+
+Running tests
+-------------
+
+Continuous Integration pipeline is based on `cargo`_.
+You can run all tests by
+
+.. code-block:: bash
+
+    cd <path/to/rascaline/repo>
+    cargo test  # or cargo test --release to run tests in release mode
+
+These are exactly the same tests that will be performed online in our
+Github CI workflows.
+You can also run only a subset of tests with one of these commands:
+
+- ``cargo test`` runs everything
+- ``cargo test --test=<test-name>`` to run only the tests in ``tests/<test-name>.rs``;
+    - ``cargo test --test=python-api`` (or ``tox`` directly) to run Python tests only;
+    - ``cargo test --test=c-api`` to run the C API tests only;
+- ``cargo test --lib`` to run unit tests;
+- ``cargo test --doc`` to run documentation tests;
+- ``cargo bench --test`` compiles and run the benchmarks once, to quickly ensure
+  they still work.
+
+You can add some flags to any of above commands to further refine which tests
+should run:
+
+- ``--release`` to run tests in release mode (default is to run tests in debug mode)
+- ``-- <filter>`` to only run tests whose name contains filter, for example ``cargo test -- spherical_harmonics``
+- ``--package rascaline`` to run tests defined in the rascaline crate (the core implementation)
+- ``--package rascaline-c-api`` to run tests defined in the rascaline-c-api
+  crate (the C API implementation)
+
+Also, you can run individual python environment tets using `Tox`_ 
+if you wish to test only specific functionalities, for example:
+
+.. code-block:: bash
+
+    tox -e lint  # code style
+    tox -e all-deps  # python tests with all dependencies
+    tox -e min-deps  # python tests with minimal dependencies
+    tox -e examples  # python tests of examples
+    tox -e build  # python packaging
+    tox -e format  # format all files
+
+The latter command `tox -e format` will use tox to do actual formatting instead
+of just testing it.
+
+Writing your own calculator
+---------------------------
+
+For adding a new calculator take a look at :ref:`dev-adding-new-calculator`.
```

Comment:
I don't think the link will work from github, should we use an absolute URL instead?